### PR TITLE
[#3807] feat(client-java): SimpleAuth can add a username when build a GravitinoClient

### PR DESCRIPTION
**Pull request by mygrsun**
### What changes were proposed in this pull request?

1.withSimpleAuth add user name as parameter to build simple auth GravitinoClient

### Why are the changes needed?

fix the https://github.com/datastrato/gravitino/issues/3807

Fix: #3807 

### Does this PR introduce _any_ user-facing change?

(Please list the user-facing changes introduced by your change, including
no

### CODE REVIEW

**Comment by jerryshao** [comment]
@qqqttt123 please help to review this.

File: docs/security.md [hunk 1]
```diff
@@ -16,6 +16,7 @@ Simple mode is the default authentication option.
 Simple mode allows the client to use the environment variable `GRAVITINO_USER` as the user.
 
 If the environment variable `GRAVITINO_USER` isn't set, the client uses the user of the machine that sends requests.
+Additionally, the username can be directly used as a parameter to create a client.
```
Comment (jerqi) [inline docs/security.md:19]: Maybe we can move this to Line29.

```
the username can be used as a parameter to create a client directly.
```

File: clients/client-java/src/main/java/com/datastrato/gravitino/client/GravitinoClientBase.java [hunk 2]
```diff
@@ -218,6 +218,17 @@ public Builder<T> withSimpleAuth() {
...
     }
 
+    /**
+     * Sets the simple mode authentication for Gravitino with a specific username
+     *
+     * @param userName The username of the user.
+     * @return This Builder instance for method chaining.
+     */
+    public Builder<T> withSimpleAuth(String userName) {
+      this.authDataProvider = new SimpleTokenProvider(userName);
```
Comment (jerqi) [inline clients/client-java/src/main/java/com/datastrato/gravitino/client/GravitinoClientBase.java:228]: Could you check the userName in the method `build`?

**Comment by yuqi1129** [comment]
@mygrsun 
Please resolve the comments and conflicts.

File: clients/client-java/src/main/java/com/datastrato/gravitino/client/GravitinoClientBase.java (hunk 2 above)
Comment (mygrsun) [reply clients/client-java/src/main/java/com/datastrato/gravitino/client/GravitinoClientBase.java:232]: the 'build' function has oauth and kerberos AuthDataProvider. so I think we should add the check in the withSimpleAuth like this.
is it ok?

File: docs/security.md (hunk 1 above)
Comment (mygrsun) [reply docs/security.md]: done

**Comment by mygrsun** [comment]
> @mygrsun Are you still working on this issue?

i have resolve the comment,please review again

File: clients/client-java/src/main/java/com/datastrato/gravitino/client/GravitinoClientBase.java (hunk 2 above)
Comment (jerqi) [reply clients/client-java/src/main/java/com/datastrato/gravitino/client/GravitinoClientBase.java]: For builder pattern, we usually check them in the build method.

**Comment by yuqi1129** [comment]
@mygrsun 
Please fix the compile issues and add some ut for the change, thanks.

**Comment by mygrsun** [comment]
> @mygrsun Please fix the compile issues and add some ut for the change, thanks.

done

**Review by yuqi1129 — APPROVED** [review]
LGTM

**Pull request merged**